afs/afs-ext-base/src/main/java/com/powsybl/afs/ext/base/AbstractScript.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2018, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.afs.ext.base;
+
+import com.google.common.io.CharStreams;
+import com.powsybl.afs.ProjectFile;
+import com.powsybl.afs.ProjectFileCreationContext;
+import com.powsybl.afs.storage.events.AppStorageListener;
+import com.powsybl.afs.storage.events.NodeDataUpdated;
+import com.powsybl.afs.storage.events.NodeEvent;
+import com.powsybl.afs.storage.events.NodeEventType;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public abstract class AbstractScript extends ProjectFile implements ReadOnlyScript {
```

Comment:
See previous remark about the name.

I don't understand why you need a read-only version of scripts? Could you give us more context?